### PR TITLE
[Sparse] Modify the test of sparse_matrix_str

### PR DESCRIPTION
**Pull request by yxy235**
## Description

Modify the test of sparse_matrix which has changed "print" to "assert".
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: tests/python/pytorch/sparse/test_sparse_matrix.py [hunk 1]
```diff
@@ -596,14 +596,30 @@ def test_print():
     col = torch.tensor([2, 1, 3]).to(ctx)
     val = torch.tensor([1.0, 1.0, 2.0]).to(ctx)
     A = from_coo(row, col, val)
-    print(A)
+    expect_res = str(
```
Comment (frozenbugs) [inline tests/python/pytorch/sparse/test_sparse_matrix.py:599]:
```suggestion
    expected = str(
```

File: tests/python/pytorch/sparse/test_sparse_matrix.py [hunk 2]
```diff
@@ -596,14 +596,30 @@ def test_print():
...
+        """SparseMatrix(indices=tensor([[1, 1, 3],
+                             [2, 1, 3]]),
+             values=tensor([1., 1., 2.]),
+             shape=(4, 4), nnz=3)"""
```
Comment (frozenbugs) [inline tests/python/pytorch/sparse/test_sparse_matrix.py:603]:
```suggestion
        """
SparseMatrix(indices=tensor([[1, 1, 3],
                             [2, 1, 3]]),
             values=tensor([1., 1., 2.]),
             shape=(4, 4), nnz=3)
        """
```

File: tests/python/pytorch/sparse/test_sparse_matrix.py [hunk 3]
```diff
@@ -596,14 +596,30 @@ def test_print():
...
     # vector-shape non zero
     row = torch.tensor([1, 1, 3]).to(ctx)
     col = torch.tensor([2, 1, 3]).to(ctx)
-    val = torch.randn(3, 2).to(ctx)
+    val = torch.tensor(
+        [[1.3080, 1.5984], [-0.4126, 0.7250], [-0.5416, -0.7022]]
+    ).to(ctx)
     A = from_coo(row, col, val)
-    print(A)
+    expect_res = str(
```
Comment (frozenbugs) [inline tests/python/pytorch/sparse/test_sparse_matrix.py:614]: ditto.

File: tests/python/pytorch/sparse/test_sparse_matrix.py (hunk 2 above)
Comment (frozenbugs) [reply tests/python/pytorch/sparse/test_sparse_matrix.py:603]: Not sure whether this passes the test or not, but looks better in this way.

**Review by frozenbugs — APPROVED** [review]

File: tests/python/pytorch/sparse/test_sparse_matrix.py (hunk 2 above)
Comment (yxy235) [reply tests/python/pytorch/sparse/test_sparse_matrix.py:603]: > Not sure whether this passes the test or not, but looks better in this way.

I tried it locally. It does not pass the test, for there will be another '\n' before the expected result, which causes assertion error.

Comment (frozenbugs) [reply tests/python/pytorch/sparse/test_sparse_matrix.py:603]: I see, then ignore this.

**Pull request merged**